src/vs/base/browser/ui/selectBox/selectBoxCustom.ts

```diff
@@ -324,13 +324,21 @@ export class SelectBoxList implements ISelectBoxDelegate, IDelegate<ISelectOptio
 
 		// Style parent select
 
+		// {{SQL CARBON EDIT}}
 		let background = null;
+		let foreground = null;
+		let border = null;
 
 		if (this.selectElement) {
-			background = this.styles.selectBackground ? this.styles.selectBackground.toString() : null;
-			const foreground = this.styles.selectForeground ? this.styles.selectForeground.toString() : null;
-			const border = this.styles.selectBorder ? this.styles.selectBorder.toString() : null;
-
+			if (this.selectElement.disabled) {
+				background = (<any>this.styles).disabledSelectBackground ? (<any>this.styles).disabledSelectBackground.toString() : null;
+				foreground = (<any>this.styles).disabledSelectForeground ? (<any>this.styles).disabledSelectForeground.toString() : null;
```

Comment:
Is there some type other than `any` that `this.styles` falls under?

Reply:
that type doesn't have `.disabledSelectBackground`

Reply:
Sure but what type is `this.styles` really in order to have `disabledSelectBackground` and `disabledSelectForeground` as properties?

Reply:
It's getting its Colors from ListBox, where `disabledSelectBackground` and `disabledSelectForeground` are properties. I don't think we can cast it into a known class, unless we introduce a new interface. I'd avoid that since it's VSCode code base, and since we've used this method before.